### PR TITLE
Options when Deleting All Rows from Table #added

### PR DESCRIPTION
**Pull request by Jason-Morcos**
## Changes:
- Added an option to delete only the selected rows even when all rows in a table are selected (instead of forcing you to run a delete all from table command)

## Closes following issues:
- Closes: #1054

## Tested:
- Processors:
  - [x] Intel
  - [ ] Apple Silicon
- macOS Versions:
  - [ ] 10.12.x (Sierra)
  - [ ] 10.13.x (High Sierra)
  - [ ] 10.14.x (Mojave)
  - [ ] 10.15.x (Catalina)
  - [x] 11.x (Big Sur)
- Localizations:
  - [x] English
  - [ ] Spanish
  - [ ] Other (please specify)
- Xcode Version:
  
## Screenshots:
Screen Shot 2021-06-01 at 16 51 16

### CODE REVIEW

**Review by Kaspik — APPROVED** [review]
Not sure about this one as there is a lot of changes and hard to review

**Comment by Jason-Morcos** [comment]
> Not sure about this one as there is a lot of changes and hard to review 

Definitely hard to follow. If on the diff page you click the little settings button and check "hide white space changes" it's a little easier to follow.
Tested all the main cases and they appeared to work (deleting a single row, deleting multiple rows, selecting all rows but choosing to only delete selected, selecting all rows and deleting all from table but not resetting auto inc, deleting all and resetting auto inc).

**Review by Kaspik — APPROVED** [review]
